yarn-project/simulator/src/public/public_processor.ts

```diff
@@ -165,7 +165,7 @@ export class PublicProcessor {
     );
     this.log.debug(`Beginning processing in phase ${phase?.phase} for tx ${tx.getTxHash()}`);
     let proof = tx.proof;
-    let publicKernelPublicInput = tx.data.toPublicKernelCircuitPublicInputs();
+    let publicKernelPublicInput = tx.data.toPublicKernelCircuitPublicInputs(this.globalVariables);
```

Comment:
This `publicKernelPublicInput` is the public inputs for `private_kernel_tail`. If we tweak the values, it will make it slightly more complicated to verify the private tail proof in `public_kernel_setup`: we will have to change the `globalVariables` back to empty in the circuit, to verify that it is indeed the public inputs of the valid proof.

I suggest that we assign the `globalVariables` to `constantData` from `PublicCircuitPublicInputs` in `public_kernel_setup`. Sort of like how we initialise the `constantData` in `private_kernel_init`, but here we only initialise/override `globalVariables`.